### PR TITLE
Add a Badge not-transferable token

### PR DESCRIPTION
**Pull request by sullof**
1. Add a Badge not-transferable token
2. Modify ByteCity.sol to manage the on-chain attributes of the badge (for now, only the level of the user)

### CODE REVIEW

**Comment by sullof** [comment]
Solves #6

File: contracts/tokens/Badge.sol [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    address _player,
+    uint256 _index
+  ) external view override returns (uint256) {
+    return _tokenAttributes[_id][_player][_index];
+  }
+
+  function initializeAttributesFor(uint256, address) external override {
+    // we allow only the city to use the badge
+    revert("Badge: only city is authorized on this badge");
+  }
```
Comment (jerrybassat7) [inline contracts/tokens/Badge.sol:68]: Is this due to one of the dependencies requiring this function to be implemented?

File: contracts/tokens/Badge.sol [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+  function updateAttributes(
+    uint256 _id,
+    uint256 _index,
+    uint256 _attributes
+  ) external override {
+    require(_tokenAttributes[_id][_msgSender()][0] != 0, "Badge: player not authorized");
+    // notice that if the playes set the attributes to zero, it de-authorize itself
```
Comment (jerrybassat7) [inline contracts/tokens/Badge.sol:76]: players*

**Review by jerrybassat7 — APPROVED** [review]
lgtm

File: contracts/tokens/Badge.sol (hunk 1 above)
Comment (sullof) [reply contracts/tokens/Badge.sol:68]: Yes

File: contracts/tokens/Badge.sol (hunk 2 above)
Comment (sullof) [reply contracts/tokens/Badge.sol:76]: Thanks

**Pull request merged**